execution_engine/src/shared/system_config/mint_costs.rs

```diff
@@ -102,7 +111,8 @@ impl ToBytes for MintCosts {
 impl FromBytes for MintCosts {
     fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), casper_types::bytesrepr::Error> {
         let (mint, rem) = FromBytes::from_bytes(bytes)?;
-        let (reduce_total_supply, rem) = FromBytes::from_bytes(rem)?;
+        let (reduce_total_supply, _) = FromBytes::from_bytes(rem)?;
+        let (burn, rem) = FromBytes::from_bytes(bytes)?;
```

Comment:
This is incorrect; `reduce_total_supply` reads from `rem` but the `rem` after `from_bytes` is put to `_`, then `burn` is deserialized from `bytes` possibly starting from scratch. I think serialization tests are not passing.